File: vigenere.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <cs50.h>
#include <ctype.h>

int main(int argc, string argv[])
{
    string k;
    string plaintext;
    if (argc != 2)
    {
        printf("Usage: ./caesar k\n");
        return 1;
    }
    else
    {
        for (int i = 0; i < strlen(argv[1]); i++)
        {
            if (!isalpha(argv[1][i]))
            {
                printf("invalid key\n");
                return 1;
            }
        }
    }

    k = argv[1];

    plaintext = get_string("plaintext: ");

    printf("ciphertext: ");
    int klen = strlen(k);

    for (int j = 0, i = 0;  i < strlen(plaintext) ; i++)
    {
        int key = toupper(k[j % klen]) - 'A';
        if (islower(plaintext[i]))
            {
                printf("%c", (plaintext[i] - 'a' + key) % 26 + 'a');
                j++;
            }
        else if (isupper(plaintext[i]))
            {
                printf("%c", (plaintext[i] - 'A' + key) % 26 + 'A');
                j++;
            }
        else printf("%c", plaintext[i]);
    }
    printf("\n");
    return 0;
}

File: vigenere.py
import cs50
import sys

#put a cypher key string in argv[1], which excepts to be a~z
def codecheck():
    if len(sys.argv)==2:
        cypherkey = sys.argv[1]
        for c in cypherkey:
            if c.isalpha() == 0:
                print("invalid key")
                return sys.exit(1)
        words = cs50.get_string("plaintext: ")
        cypher(cypherkey, words)
    else:
        print("usage: python vigenere.py key")
        return sys.exit(1)
def cypher(cypherkey, words):
    wordslen = len(words)
    cypherwords=''
    j = 0
    for i in range(wordslen):
        #repeat the key you parse into
        key = ord(cypherkey[j%len(cypherkey)].upper())-65

        if words[i].islower() == 1:
            cypherwords += chr((ord(words[i])-97+key)%26+97)
            j+=1
        elif words[i].isupper() == 1:
            cypherwords += chr((ord(words[i])-65+key)%26+65)
            j+=1
        else:
            cypherwords += words[i]
    print(f"ciphertext: {cypherwords}")


if __name__ == "__main__":
    codecheck()